### ISSUE TITLE
Node sass fails to install (io.js 3.0.0 / osx)

### ISSUE THREAD

**Issue by teeli** [sass/node-sass#1086]
node-sass seems to throw a bunch or node-gyp errors while installing on io.js 3.0.0 on osx.

I see multiple errors like:
 `/Users/teel/.node-gyp/3.0.0/deps/v8/include/v8.h:221:5: error: assigning to 'v8::Function *volatile' from incompatible type 'v8::Value *'`

```
4 warnings and 10 errors generated.
make: *** [Release/obj.target/binding/src/binding.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/teel/Projects/redacted/node_modules/node-sass/node_modules/pangyp/lib/build.js:271:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.4.0
gyp ERR! command "/usr/local/bin/iojs" "/Users/teel/Projects/redacted/node_modules/node-sass/node_modules/pangyp/bin/node-gyp" "rebuild" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /Users/teel/Projects/redacted/node_modules/node-sass
gyp ERR! node -v v3.0.0
gyp ERR! pangyp -v v2.3.0
gyp ERR! not ok 
Build failed
npm ERR! Darwin 14.4.0
npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "install" "sass-loader"
npm ERR! node v3.0.0
npm ERR! npm  v2.13.5
npm ERR! code ELIFECYCLE

npm ERR! node-sass@3.2.0 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-sass@3.2.0 postinstall script 'node scripts/build.js'.
npm ERR! This is most likely a problem with the node-sass package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/build.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls node-sass
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/teel/Projects/redacted/npm-debug.log
```

**Comment by AbrahamAlcaina** [comment]
I have the same issue

**Comment by chncdcksn** [comment]
Same error. Ubuntu 15.04, io.js 3.1.0

https://gist.github.com/chncdcksn/f72d62d7ad13a400312a

**Comment by dhritzkiv** [comment]
Ditto. OS X 10.10; io.js 3.1.0; node-gyp 3.1.0

**Comment by saper** [comment]
This is #1053, currently fixed in git master and under testing for a release.

**Comment by ai** [comment]
Same problem in PostCSS benchmark.

**Comment by saper** [comment]
> Same problem in PostCSS benchmark.

See #1053 and #1054.

**Comment by ai** [comment]
@saper latest `node-sass` still falls :)

**Comment by saper** [comment]
> @saper latest `node-sass` still falls :)

Try git master.

**Comment by jbardrof** [comment]
It still appears to be failing:

`npm install git+https://git@github.com/sass/node-sass.git`

`Cannot download "https://github.com/sass/node-sass/releases/download/v3.2.0/linux-x64-45_binding.node": HTTP error 404 Not Found`

**Comment by saper** [comment]
This is because #1054 is not released yet.

**Comment by mansona** [comment]
Is there an ETA on that release?

**Comment by xzyfer** [comment]
This will likely ship in the AM AEST (approx 14hrs.).

**Comment by frankleng** [comment]
+1 to get updates. thanks

**Comment by mansona** [comment]
@xzyfer thanks for the update, I look forward to the release :+1:

**Comment by xzyfer** [comment]
[v3.3.0 has been released](https://github.com/sass/node-sass/releases/tag/v3.3.0)

**Comment by xzyfer** [comment]
Reopening this because we've had to pull the v3.3.x release due a segfault issue when used with fsevent (https://github.com/sass/node-sass/issues/1107).

**Comment by xzyfer** [comment]
If you desperately need iojs v3 support you can use the beta tag, but we caution against this.

```
npm install node-sass@beta
```

**Comment by jalcine** [comment]
[**@xzyfer**](https://github.com/xzyfer) commented on [Aug 26, 2015, 9:58 AM EDT](https://github.com/sass/node-sass/issues/1086#issuecomment-135029281 "2015-08-26T13:58:11Z - Replied by Github Comment Enhancer"):
> If you desperately need iojs v3 support you can use the beta tag, but we caution against this.
> 
>     npm install node-sass@beta

So far, I haven't run into any issues; but I'll be sure to report them (I'm using this via broccoli).

**Comment by puzrin** [comment]
@xzyfer any info when 3.3.1+ will be available in npm as stable (not as `node-sass@beta`)?

**Comment by saper** [comment]
We need to check why https://github.com/strongloop/fsevents/issues/82 occurs in conjunction with node-sass 3.3.0

**Comment by xzyfer** [comment]
@puzrin there's not estimate yet. We're having an incompatiblity with `fsevents` which is widely used and core to projects like gulp and webpack. Until this is resolved we cannot ship iojs v3 compatibility.

Subscribe to https://github.com/sass/node-sass/issues/1107 and strongloop/fsevents#82 for updates on the situation.

**Comment by puzrin** [comment]
No prob. I have broken travis tests in `mincer` and wished to understand, should i disable iojs env or wait a bit. If `fsevent` is the only blocker, there are a good probability of improvement in near future.

**Comment by xzyfer** [comment]
Yes, we're purely blocked on fsevents at the moment.

You can enable iojs for specific versions in travis like do https://github.com/sass/node-sass/blob/master/.travis.yml#L7-L13

Just remove `iojs-v3` and you should be fine to use Node Sass.

**Comment by markogresak** [comment]
Please note that installing `@beta` will throw segfault even in node v0.12.7, happened to me after switching from v3.0.0 (also segfaulted). It happened in a project where I was using [sass-loader](https://github.com/jtangelder/sass-loader).

**Comment by xzyfer** [comment]
Fixed in v3.3.2.

**Comment by boiawang** [comment]
update gcc(4.6+).
